### PR TITLE
fix software errors/warnings

### PR DESCRIPTION
**Pull request by davideschiavone**

### CODE REVIEW

File: sw/device/lib/drivers/rv_plic/rv_plic.c [hunk 1]
```diff
@@ -371,7 +371,7 @@ plic_result_t plic_assign_external_irq_handler( uint32_t id,
 {
   if( id >= EXT_IRQ_START && id <= QTY_INTR )
   {
-    handlers[ id ] = (handler_funct_t*) handler;
+    handlers[ id ] = (handler_funct_t) handler;
```
Comment (JoseCalero) [inline sw/device/lib/drivers/rv_plic/rv_plic.c:374]: @StefanoAlbini96 can you check this tomorrow, please? (after cr ;D)

**Review by JoseCalero — COMMENTED** [review]
asking @StefanoAlbini96 to check the plic change

**Review by StefanoAlbini96 — APPROVED** [review]
approve changes

**Comment by StefanoAlbini96** [comment]
ready to be merged

**Pull request merged**